app/ts/components/subcomponents/ChainSelector.tsx

```diff
@@ -27,17 +28,49 @@ export function ChainSelector(params: ChainSelectorParams) {
 			</button>
 		</div>
 		<div class = 'dropdown-menu' id = 'dropdown-menu' role = 'menu' style = { { left: 'unset' } }>
-			<div class = 'dropdown-content' style = { { position: 'fixed' } }>
-				{
-					params.rpcEntries.value.map((rpcEntry) => {
-						return (
-							<button type = 'button' class = { `dropdown-item ${ params.rpcNetwork.value !== undefined && rpcEntry.httpsRpc === params.rpcNetwork.value.httpsRpc ? 'is-active' : '' }` } onClick = { () => changeRpc(rpcEntry) } >
-								{ rpcEntry.name }
-							</button>
-						)
-					})
-				}
-			</div>
+			<div class = 'dropdown-content' style = { { position: 'fixed' } }> {
+				params.rpcEntries.value.map((rpcEntry) => <>
+					<button type = 'button' class = { `dropdown-item ${ params.rpcNetwork.value !== undefined && rpcEntry.httpsRpc === params.rpcNetwork.value.httpsRpc ? 'is-active' : '' }` } onClick = { () => changeRpc(rpcEntry) } >
+						{ rpcEntry.name }
+					</button>
+				</>)
+			} </div>
+		</div>
+	</div>
+}
+
+interface ChainSelectorParams {
+	chainId: ReadonlySignal<ChainIdWithUniversal>
+	rpcEntries: Signal<RpcEntries>
+	changeChain: (entry: ChainEntry) => void
+}
+
+export function ChainSelector(params: ChainSelectorParams) {
+	const [isOpen, setIsOpen] = useState(false)
```

Comment:
Wouldn't be much to switch to signal as the entire component already is utilizing it already anyway

```ts
// const [isOpen, setIsOpen] = useState(false)
const dropdownOpen = useSignal(false)
```